arch-update/arch-update.py

```diff
@@ -59,17 +60,29 @@ def get_updates():
     return updates
 
 
-def get_aur_updates():
+def get_aur_updates(aur_helper):
     output = ''
-    try:
-        output = check_output(['yaourt', '-Qua']).decode('utf-8')
-    except subprocess.CalledProcessError as exc:
-        # yaourt exits with 1 and no output if no updates are available.
-        # we ignore this case and go on
-        if not (exc.returncode == 1 and not exc.output):
-            raise exc
-    if not output:
-        return []
+
+    if aur_helper == 'yaourt':
+        try:
+            output = check_output(['yaourt', '-Qua']).decode('utf-8')
+        except subprocess.CalledProcessError as exc:
+            # yaourt exits with 1 and no output if no updates are available.
+            # we ignore this case and go on
+            if not (exc.returncode == 1 and not exc.output):
+                raise exc
+        if not output:
+            return []
+
+    elif aur_helper == 'yay':
+        try:
+            output = check_output(['yay', '-Qua']).decode('utf-8')
+        except subprocess.CalledProcessError as exc:
+            # yay also exits with 1 and no output if no updates are available
+            if not (exc.returncode == 1 and not exc.output):
+                raise exc
+        if not output:
+            return []
 
```

Comment:
It is perhaps a wise idea to DRY the code a bit here.

The only difference between "yaourt" and "yay" is the executable (and maybe the arguments).

The entire logic to prepare the command, run it, catch exceptions shouldn't be duplicated (imagine if we had 10 AUR helpers)